Samples/WindowsCopilotRuntime/cs-maui/MauiWCRSample/Platforms/Windows/MainPage.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Devices.Scanners;
+using Microsoft.Graphics.Imaging;
+using Microsoft.Maui.Platform;
+using Microsoft.UI.Xaml.Media.Imaging;
+using Windows.Graphics.Imaging;
+using Windows.Storage.Streams;
+using Microsoft.Windows.AI.Generative;
+
+namespace MauiWindowsCopilotRuntimeSample // extending the partial class MainPage with the platform-specific implementation
+{
+    partial class MainPage
+    {
+        partial void ShowImageScalingPanelIfSupported()
+        {
+            ShowImageScalingPanelIfSupportedImpl();
+        }
+
+        private async void ShowImageScalingPanelIfSupportedImpl()
+        {
+            try
+            {
+                if (!ImageScaler.IsAvailable())
+                {
+                    System.Diagnostics.Debug.WriteLine("Calling ImageScaler.MakeAvailableAsync");
+                    var op = await ImageScaler.MakeAvailableAsync();
+                    System.Diagnostics.Debug.WriteLine($"ImageScaler.MakeAvailableAsync completed with Status: {op.Status}");
+                }
+
+                if (ImageScaler.IsAvailable())
+                {
+                    // Image scaling is supported, so show the content
+                    imageScalingPanel.IsVisible = true;
+                }
+            }
+            catch (Exception ex)
+            {
+                // Log the exception
+                System.Diagnostics.Debug.WriteLine("Unable to use ImageScaler: " + ex.Message);
+            }
+        }
+
+        partial void ScaleImage(double scale)
+        {
+            DoScaleImage(scale);
+        }
+
+        private async void DoScaleImage(double scale)
+        {
+            // Load the original image
+            var resourceManager = new Microsoft.Windows.ApplicationModel.Resources.ResourceManager();
```

Comment:
Since your project has Android and other platforms, does this need to be under some WINDOWS_ONLY check or is this already abstracted out? i.e., with the other platform builds fail at compile/runtime? Should there be some IsWindows() runtime check or something?

Reply:
It is true that `#if WINDOWS` blocks could be used to ensure code in those blocks are only used in the build for Windows target. But I found that approach to be a poor choice for anything more than a line or two of trivial code, since that code is gray-ed out (VS sees the #define isn't always set) and there is no IntelliSense/auto-suggest inside those blocks.

What this sample uses instead is MAUI's other platform-specific code option: defining partial classes with partial methods and implementing platform-specific code in the platform-specific versions of those partial classes. This `Platforms/Windows/MainPage.cs` is the Windows-specific implementation of the partial class, so no `#if WINDOWS` lines are needed.

Reply:
It doesn't _have_ to be a partial class, any code files you put inside the `Platform/PlatformName` folder, it wil automatically be compiled only for that target platform. Functionally identical to using `#IF platformname` anywhere else in the project..

Using a partial class for MainPage.xaml.cs just makes it more convenient to access those methods... however, I urge a little bit of caution here. It isn't obvious that the logic is within the Platform/Windows folder, so I would add a comment to the original MainPage.xaml/MainPage.xaml.cs to the reader to let them know where to look for the rest of the partial class.